### DISCUSSION TITLE
Understanding of the Adonis workflow

### DISCUSSION THREAD

**Discussion by woodgates** [adonisjs/core#1523] [Help] [answered]
Hi.

I've been using Adonis for a while and now I'm facing some problems which I'm not sure if they should happen or it's just the way it works.

I have the following models: Product and Tags. A Product can have many Tags and they are related by the products_tags table.

Every time tags change I update Products whith the following code:

ProductController.js

```
update({ ...}){

  const {tags} = request.all()

  const product = await Product.find(params.id)

  await product.tags().detach()

  const tagsAdded = await product.addTags(tags)     // a Trait that format tags, eliminates duplicates, creates new tags and attaches tags to product. Returns attached tags 


   return tagsAdded

}
```

Due to a problem in Vuetify, two requests are sent almost intermediately when you create a duplicated tag in the Combobox component. 

So Adonis receives request1 and request2 separeted by milliseconds.

What happens with my code is the following:

1. request1 is received
2. tags are detached
3. request2 is received
4. tags are detached
5. request1 is processed by await product.addTags(tags)
6. request2 is processed by await product.addTags(tags)
7. ERROR: duplicated products_tags are created.

This is the way it should work? I ask because I thought that request1 should be processed all over and then request 2 will be processed, something like this:

request1 is received
request1 tags are detached
request2 is received
request1 is processed by await product.addTags(tags)
request1 response is sent
request2 tags are detached
request2 is processed by await product.addTags(tags)
request2 response is sent
NO ERROR

Please make comments about this.

**Comment by thetutlage** [comment]
Why you expect the server to handle requests sequentially?

**Reply by woodgates** [reply]
I just assumed it was the way it works.

I didn't have problems with this until now that I'm implementing Tags and a Gallery upload. This last one sends several files and my code wasn't working as I though I will. 

I wanted to know if I have a problem with my code (some async await wrong or something like that) or if its the way it works.

I resolved this issue this way:
- I put a debounce function on the Vuetify component to handle the request send to stop this double request.
- I moved the detach() to the addTags() Trait just before the attach(tagsIdsArray).

My Best.

**Comment by popovicieduard** [comment, marked as answer]
You can make use of transactions to persist your relationships without worrying about the fact that another request may interfere with the other request.

**Reply by woodgates** [reply]
thanks for the suggestion.

**Reply by fnoquiq** [reply]
Search for ACID
